Publications/NewsTableViewController.swift

```diff
@@ -34,41 +62,51 @@ class NewsTableViewController: UITableViewController {
-            cell.postTitleLabel.text = "There was theater performance. Everyone was great"
-            cell.authorLabel.text = "Sarah Trop"
-            cell.authorImageView.image = UIImage(named: "article")
+        if arrayOfArticle[indexPath.row].cell == 1 {
+            let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath) as! NewsTableViewCell
+            cell.authorLabel.text = arrayOfArticle[indexPath.row].author
+            cell.postTitleLabel.text = arrayOfArticle[indexPath.row].title
+            cell.postImageView.image = arrayOfArticle[indexPath.row].articleImage
+            cell.authorImageView.image = arrayOfArticle[indexPath.row].userImage
+            cell.timeStamp.text = arrayOfArticle[indexPath.row].time
             
-        } else if (indexPath as NSIndexPath).row == 1 {
-            cell.postImageView.image = UIImage(named: "article")
-            cell.postTitleLabel.text = "A cool person made a beautiful and intriguing art gallery"
-            cell.authorLabel.text = "Author"
-            cell.authorImageView.image = UIImage(named: "article")
+            return cell
             
-        } else if (indexPath as NSIndexPath).row == 2 {
-            cell.postImageView.image = UIImage(named: "article")
-            cell.postTitleLabel.text = "A cool person made a beautiful and intriguing art gallery"
-            cell.authorLabel.text = "Author"
-            cell.authorImageView.image = UIImage(named: "article")
+        } else if arrayOfArticle[indexPath.row].cell == 2 {
+            let cell = tableView.dequeueReusableCell(withIdentifier: "TextCell", for: indexPath) as! TextNewsTableViewCell
+            cell.userName.text = arrayOfArticle[indexPath.row].author
+            cell.articleLabel.text = arrayOfArticle[indexPath.row].title
+            cell.profilePic.image = arrayOfArticle[indexPath.row].userImage
+            cell.timeStamp.text = arrayOfArticle[indexPath.row].time
+            
+            return cell
             
         } else {
-            cell.postImageView.image = UIImage(named: "article")
-            cell.postTitleLabel.text = "This article is about sports"
-            cell.authorLabel.text = "Author"
-            cell.authorImageView.image = UIImage(named: "article")
+            let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath) as! NewsTableViewCell
+            cell.authorLabel.text = arrayOfArticle[indexPath.row].author
+            cell.postTitleLabel.text = arrayOfArticle[indexPath.row].title
+            cell.postImageView.image = arrayOfArticle[indexPath.row].articleImage
+            cell.authorImageView.image = arrayOfArticle[indexPath.row].userImage
+            cell.timeStamp.text = arrayOfArticle[indexPath.row].time
             
+            return cell
         }
         
-        return cell
-}
+    }
+    
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
```

Comment:
// Restrict height of cell depending on whether or not image feature is used. 
// Else case will return 94 if it becomes case no image